portal-web/test/functional/com/liferay/portalweb/macros/json/layoutpagetemplate/JSONLayoutpagetemplate.macro

```diff
@@ -34,6 +34,11 @@ definition {
 					fileEntryTypeName = ${subType},
 					groupId = ${globalGroupId});
 			}
+			else if ("${subType}" == "Basic Document") {
+				var classTypeId = JSONDLfileentrytype._getFileEntryTypeId(
+					fileEntryTypeName = "Basic Document",
+					groupId = "0");
+			}
```

Comment:
I think `JSONDLfileentrytype._getFileEntryTypeId` will need to be updated as well. This PR makes the basic document type company scoped (i.e. there will be 1 distinct basic document type per company), so you need to pass the companyId to get the correct one.

Reply:
Note that this may work if the JSONWS infrastructure sets the correct company ID in the context (I think it will do it based on the virtual host name), but you could also get weird results; the group 0 is valid in all companies, so you could get a basic doc type from another company.

Reply:
This will probably require backend changes. Maybe I could expose the `getBasicDocumentType(companyId)` method? That could make your life easier.

Reply:
Yes that would be perfect @adolfopa !

Reply:
Ok. I'll change that in my branch so you can cherry pick your commit on top.